### PR TITLE
Make Multi#initialize take 2 args, add indent opts

### LINKED ISSUE #17: Multi Spinner documentation
`TTY::Spinner::Multi` needs documentation

- [x] Document the methods in code
- [x] Add docs to README
- [x] Ensure there is an explanation of how passing options to `TTY::Spinner::Multi` will in turn affect spinners created with `#register`

### LINKED ISSUE #18: Displaying multi spinners in a tree format
Ability to display spinners in a tree like format. By default, if the multispinner is created with top level spinner, all the child spinners are printed in a tree like format. However, if no top level spinner is provided, all spinners are displayed without any decoration. The formatting can follow similar symbols to [tty-tree](https://github.com/piotrmurach/tty-tree#1-usage)

The format can be configured by providing the following option:
```ruby
indent: 2, # indentation depth
style: { # formatting style
  top: '--',
  mid: '|',
  bottom: '--'
}
```

### LINKED ISSUE #19: Make multi spinner initialization consistent with single spinner
`TTY::Spinner::Multi#initialize` should take arguments in the same form as `TTY::Spinner#initialize`

### PR DESCRIPTION
**Pull request by austb**
Fixes #18. Fixes #19.

Also Fixes #17 now

### CODE REVIEW

File: lib/tty/spinner/multi.rb [hunk 1]
```diff
@@ -17,9 +17,53 @@ class Multi
...
+      # Initialize the indentation options and remove them from the default
+      # spinner options
+      #
+      # @api private
+      def init_indent_opts
+        @indentation_opts = {}
+        @indentation_opts[:indent] = @options.delete(:indent) { 2 }
+        @indentation_opts[:style] = @options.delete(:style) {
+          {
+            top: '',
```
Comment (austb) [inline lib/tty/spinner/multi.rb]: The "coverage decrease" is being cause by this line being counted as "relevant" but then not  "tested" even when there's a test covering the default case.

**Comment by austb** [comment]
@piotrmurach What do you think about this?

**Comment by austb** [comment]
@piotrmurach thanks so much for your work on this! Any chance you can release this soon? We're going to build release packages soon and we can only build from published gem sources. Thanks again, this is going to be a great UX experience :)

**Comment by piotrmurach** [comment]
@austb thanks, enjoyed all the innovation you have brought to the library! I'm kind of pressed for time as I'm flying end of the week and got few things to get done beforehand, so could you let me know when you planning to cut your release and I will make sure I've got something out for you? Are you using the master branch in your codebase before release to test if all is fine?

**Comment by austb** [comment]
@piotrmurach We are shipping the release on Tuesday, August 15, so ideally we would have the gems published by this Friday (August 11), but we can make Sunday (August 13th) work as well. If you want to do it as a .pre release so others don't pull it in yet, that works for us as well. Thanks!

**Comment by piotrmurach** [comment]
@austb I've succeeded releasing `tty-spinner v0.5.0` & `tty-prompt v0.13.0` that fixes windows keypress reading! You should be able to cut your release now. I'm gonna be rather hard to catch this weekend, just so you know. But then again, I'm confident with the latest releases 😄

**Comment by austb** [comment]
@piotrmurach  They look great, we already had stuff running against the updated versions, so they work great :) have a great weekend! And thank you so much for all your work with these gems. They are fantastic.

**Pull request merged**